lib/gettext.nit

```diff
@@ -0,0 +1,80 @@
+# Internationalization of Strings using `gettext` library
+module gettext is
+	new_annotation language
+end
+
+in "C header" `{
+	#include <libintl.h>
+`}
+
+class PObject
+	var locations: Array[String]
+	var msgid: String is writable
+	var msgstr: String is writable
+end
+
+class POFile
+	super Writable
+
+	# Map of the strings's `msgid` and `msgstr`
+	#
+	# Read from a PO file
+	var strings: Array[PObject]
+
+	init with_strings(sm: Array[PObject])do
+		strings = new Array[PObject].with_capacity(sm.length)
+		strings.add_all sm
+	end
+
+	redef fun write_to_file(path) do
+		if not path.has_suffix(".po") then path += ".po"
+		var ofs = new FileWriter.open(path)
+		write_to(ofs)
+		ofs.close
+	end
+
+	redef fun write_to(ofs) do
+		for i in strings do
+			ofs.write("#: {i.locations.join(", ")}\n")
+			ofs.write("msgid \"{i.msgid}\"\n")
+			ofs.write("msgstr \"{i.msgstr}\"\n\n")
+		end
+	end
+
+	# Writes the information of the POFile to a .pot template file
+	fun write_template(path: String) do
+		if not path.has_suffix(".pot") then path += ".pot"
+		var f = new FileWriter.open(path)
+		write_to(f)
+		f.close
+	end
+end
+
+fun get_text_domain_foldr(s, domain, dir: String): String do
+	setdirdomain(domain, dir)
+	settextdomain(domain)
+	return gettext(s)
+end
+
+# Gets the translated text for `s` with default domain name
+#
+# Default domain name is supposed to be set once in the program or changed via
+# textdomain in C.
+fun gettext(s: String): String is extern import String.to_cstring, NativeString.to_s `{
```

Comment:
Good call for String as receiver !

I was just doing a 1-1 transformation of gettext to get it working, I just forgot to include them in String.